src/app/category/[categoryName]/page.tsx

```diff
@@ -0,0 +1,39 @@
+import { fetchProducts } from "@/lib/Utils/Api";
+import Link from "next/link";
+type Product = {
+    id: number;
+    title: string;
+    price: number;
+    description: string;
+    category: string;
+    image: string;
+};
+
+export default async function ProductListPage({ params }: { params: { categoryName: string } }) {
+    const products = await fetchProducts<Product[]>(`/products?category=${params.categoryName}`);
+
+    return (
+        <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6 p-6 sm:p-10 bg-gray-100 min-h-screen">
```

Comment:
Try minimizing like this
   @layer components {
     .button {
       @apply rounded p-1 bg-gray-200 text-black;
     }
   }